pkg/operators/klusterlet/controllers/statuscontroller/klusterlet_status_controller.go

```diff
@@ -126,6 +147,39 @@ type klusterletAgent struct {
 	getSSARFunc    getSelfSubjectAccessReviewsFunc
 }
 
+// Check both agent deployments, if the desired replicas is equal to available replicas, return available condition
+func checkAgentsDeployment(ctx context.Context, kubeClient kubernetes.Interface, agents []klusterletAgent) metav1.Condition {
+	availableMessages := []string{}
+	for _, agent := range agents {
+		deployment, err := kubeClient.AppsV1().Deployments(agent.namespace).Get(ctx, agent.deploymentName, metav1.GetOptions{})
+		if err != nil {
+			return metav1.Condition{
+				Type:    klusterletProgressing,
+				Status:  metav1.ConditionTrue,
```

Comment:
I think this should return Available=false

Reply:
So, how we define the `Progressing` status?

Reply:
progressing is a status that deployment is not fully  up, or being rolled out.

Reply:
Thanks for clarifying this, then, I'll only make the enhancement for implementation of the `Available` condition in this PR.